src/Denudation/Abstract.jl

```diff
@@ -14,7 +14,7 @@ abstract type DenudationType end
 FIXME Computes the denudation for a single time-step, given denudation parameters `param` and a simulation state `state`. `param` should have a `DenudationType` type and `state` should contain the `height` property and `sealevel`.
 """
 function denudation(input)
-    denudation_mass::Array{typeof(1.0u"m/kyr"),2} = Array{typeof(1.0u"m/kyr")}(undef, input.box.grid_size...)
+    denudation_mass::Union{Array{typeof(1.0u"m/kyr"),2}, Nothing} = Array{typeof(1.0u"m/kyr")}(undef, input.box.grid_size...)
```

Comment:
We're trying to prevent unneeded memory allocations. Here, what you're saying is: I have this variable `denudation_mass` which is either an array or nothing, and then you assign a big empty array to it. That doesn't make much sense.

Reply:
right I can initialise with nothing...